gsplat/experimental/cuda/__init__.py

```diff
@@ -0,0 +1,146 @@
+import math
+
+import torch
+
+from ._torch_impl import (
+    _isect_offset_encode,
+    _isect_tiles,
+    _persp_proj,
+    _projection,
+    _quat_scale_to_covar_perci,
+    _world_to_cam,
+    accumulate,
+)
+from ._warpper import (
+    isect_offset_encode,
+    isect_tiles,
+    persp_proj,
+    projection,
+    quat_scale_to_covar_perci,
+    rasterize_to_indices_iter,
+    rasterize_to_pixels,
+    world_to_cam,
+)
+
+
+def rendering(means, quats, scales, opacities, colors, viewmats, Ks, width, height):
+    tile_size = 16
+    C = len(viewmats)
+    covars, _ = quat_scale_to_covar_perci(quats, scales, compute_perci=False, triu=True)
+    radii, means2d, depths, conics = projection(
+        means, covars, viewmats, Ks, width, height
+    )
+    tile_width = math.ceil(width / tile_size)
+    tile_height = math.ceil(height / tile_size)
+    tiles_per_gauss, isect_ids, gauss_ids = isect_tiles(
+        means2d, radii, depths, tile_size, tile_width, tile_height
+    )
+    isect_offsets = isect_offset_encode(isect_ids, C, tile_width, tile_height)
+    render_colors, render_alphas = rasterize_to_pixels(
+        means2d,
+        conics,
+        colors,
+        opacities,
+        width,
+        height,
+        tile_size,
+        isect_offsets,
+        gauss_ids,
+    )
+    return render_colors, render_alphas
+
+
+def _rendering(means, quats, scales, opacities, colors, viewmats, Ks, width, height):
```

Comment:
is the `_` naming convention in this file meant to follow what you have in `_torch_impl` where all the native pytorch methods are led with underscores and the cuda methods aren't? It looks like this function might be meaning to test out the pytorch methods which are imported but not called

Reply:
I need to cleanup `__init__.py` and define these functions somewhere else and rename them properly. For this `_rendering` function, the current usage of it is to test against the `rasterize_to_pixels` CUDA implementation only.   So functions other than that part are still using the CUDA implementation to minimize the numerical difference for the test. But I do plan to make all other functions in it using torch impl functions in the end.